### ISSUE TITLE
Issues when chaining withText and findReact

### ISSUE THREAD

**Issue by aabde** [DevExpress/testcafe-react-selectors#97]
Component structure :
```xml
<List>
  <Item>
    {'Henry'}
    <Description>
      {'amazing man'}
    </Description>
  </Item>
  <Item>
    {'Charles'}
    <Description>
      {'random guy'}
    </Description>
  </Item>
</List>
```
tests:
```js
await ReactSelector('Item').count;
// returns 2 as expected

await ReactSelector('Item').withText('Charles').innerText;
// returns 'Charles' as expected

await  ReactSelector('Item').withText('Charles').findReact('Description');
// returns 'amazing man' but should return 'random guy'
```

When chaining `withText` and `findReact`, the findReact does not search in nodes matching the withText results but in all the matching nodes ignoring the withText.

**Comment by kirovboris** [comment]
Hi @aabde, 

Thanks for reporting it. I have reproduced this issue and I'll investigate it soon.

**Comment by Lukas-Kullmann** [comment]
What's the status of this? I have the same issue with `nth` instead of `withText`, too.

**Comment by kirovboris** [comment]
I'm working on this. I've found the reason of this bug and I'm going to fix it this week.

**Comment by kirovboris** [comment]
I've just published `testcafe-react-selector@2.0.1`.
